rwengine/src/audio/SoundManager.cpp

```diff
@@ -540,88 +87,170 @@ bool SoundManager::loadSound(const std::string& name,
                                        std::forward_as_tuple());
         sound = &emplaced.first->second;
 
-        sound->source.loadFromFile(path);
-        sound->isLoaded = sound->buffer.bufferData(sound->source);
+        sound->source = std::make_shared<SoundSource>();
+        sound->buffer = std::make_unique<SoundBuffer>();
+
+        sound->source->loadFromFile(fileName);
+        sound->isLoaded = sound->buffer->bufferData(*sound->source);
     }
 
     return sound->isLoaded;
 }
-bool SoundManager::isLoaded(const std::string& name) {
-    if (sounds.find(name) != sounds.end()) {
-        return sounds[name].isLoaded;
+
+void SoundManager::loadSfxSound(const rwfs::path& path, size_t index) {
+    Sound* sound = nullptr;
+
+    auto emplaced =
+        sfx.emplace(std::piecewise_construct, std::forward_as_tuple(index),
+                    std::forward_as_tuple());
+    sound = &emplaced.first->second;
+
+    sound->source = std::make_shared<SoundSource>();
+    sound->source->loadSfx(path, sdt, index);
+}
+
+size_t SoundManager::createSfxInstance(size_t index) {
+    Sound* sound = nullptr;
+    auto soundRef = sfx.find(index);
```

Comment:
Here we can check, if sound is loaded, and if there's need load it. But it would require storing game path in SoundManager.

Reply:
To do later.